apps/web/src/server/views/applications/case/components/case-details-table.component.njk

```diff
@@ -1,30 +1,35 @@
 {% from "govuk/components/table/macro.njk" import govukTable %}
 {% from "./case-edit-link.component.njk" import caseEditLink %}
 
-{% macro caseDetailsTable(case, isPublishPage, isWelsh) %}
+{% macro caseDetailsTable(params, isPublishPage, isWelsh) %}
 	{% set regionNames = [] %}
-	{% for item in case.geographicalInformation.regions %}
+	{% for item in params.case.geographicalInformation.regions %}
 		{% set regionNames = (regionNames.push(item.displayNameEn), regionNames) %}
 	{% endfor %}
 
 	{% set gridReferences = '' %}
-	{% if case.geographicalInformation.gridReference.easting %}
-		{% set gridReferences = case.geographicalInformation.gridReference.easting + ' (Easting)' %}
+	{% if params.case.geographicalInformation.gridReference.easting %}
+		{% set gridReferences = params.case.geographicalInformation.gridReference.easting + ' (Easting)' %}
 	{% endif %}
 	{% set gridReferences = gridReferences + '<br>' %}
-	{% if case.geographicalInformation.gridReference.northing %}
-		{% set gridReferences = gridReferences + case.geographicalInformation.gridReference.northing + ' (Northing)' %}
+	{% if params.case.geographicalInformation.gridReference.northing %}
+		{% set gridReferences = gridReferences + params.case.geographicalInformation.gridReference.northing + ' (Northing)' %}
 	{% endif %}
 
-  {% set rowsData = buildCaseInformation({ case: case, gridReferences: gridReferences, regionNames: regionNames }, isWelsh) %}
+  {% set rowsData = buildCaseInformation({
+    case: params.case,
+    keyMembers: params.keyMembers,
+    gridReferences: gridReferences,
+    regionNames: regionNames
+  }, isWelsh) %}
 
 	{% set tableRows = [] %}
 	{% for rowData in rowsData %}
 		{#  In the value column, rowData can be used because "title" and "url" are ignored	#}
 		{% set cols = [{text: rowData.title}, rowData] %}
 
 		{% set linkCol =  {
```

Comment:
```suggestion
		{% if not isPublishPage %}
			{% if rowData.url %}
				{% set linkCol =  {
					html: caseEditLink({editStepUrl:rowData.url, caseId: params.case.id}),
					classes: 'text-align--right pins-table__cell--s'
				} %}
                        {% else %}
				{% set linkCol =  {} %}
                        {% endif %}
			{% set cols = (cols.push(linkCol), cols) %}
		{% endif %}
```

Reply:
As above, I think the link should not be added if it doesn't exist in the rowData